tests/components/humidifier/test_device_condition.py

```diff
@@ -0,0 +1,314 @@
+    )
+    entity_reg.async_get_or_create(DOMAIN, "test", "5678", device_id=device_entry.id)
+    hass.states.async_set(
+        f"{DOMAIN}.test_5678",
+        STATE_ON,
+        {
+            const.ATTR_MODE: const.MODE_AWAY,
+            const.ATTR_AVAILABLE_MODES: [const.MODE_HOME, const.MODE_AWAY],
+        },
+    )
+    hass.states.async_set(
+        "humidifier.test_5678", "attributes", {"supported_features": 1}
+    )
+    expected_conditions = [
+        {
+            "condition": "device",
+            "domain": DOMAIN,
+            "type": "is_off",
+            "device_id": device_entry.id,
+            "entity_id": f"{DOMAIN}.test_5678",
+        },
+        {
+            "condition": "device",
+            "domain": DOMAIN,
+            "type": "is_on",
+            "device_id": device_entry.id,
+            "entity_id": f"{DOMAIN}.test_5678",
+        },
+        {
+            "condition": "device",
+            "domain": DOMAIN,
+            "type": "is_mode",
+            "device_id": device_entry.id,
+            "entity_id": f"{DOMAIN}.test_5678",
+        },
+    ]
+    conditions = await async_get_device_automations(hass, "condition", device_entry.id)
+    assert_lists_same(conditions, expected_conditions)
+
+
+async def test_get_conditions_toggle_only(hass, device_reg, entity_reg):
+    """Test we get the expected conditions from a humidifier."""
+    config_entry = MockConfigEntry(domain="test", data={})
+    config_entry.add_to_hass(hass)
+    device_entry = device_reg.async_get_or_create(
+        config_entry_id=config_entry.entry_id,
+        connections={(device_registry.CONNECTION_NETWORK_MAC, "12:34:56:AB:CD:EF")},
+    )
+    entity_reg.async_get_or_create(DOMAIN, "test", "5678", device_id=device_entry.id)
+    hass.states.async_set(
+        f"{DOMAIN}.test_5678",
+        STATE_ON,
+        {
+            const.ATTR_MODE: const.MODE_AWAY,
+            const.ATTR_AVAILABLE_MODES: [const.MODE_HOME, const.MODE_AWAY],
+        },
+    )
+    hass.states.async_set(
+        "humidifier.test_5678", "attributes", {"supported_features": 0}
+    )
+    expected_conditions = [
+        {
+            "condition": "device",
+            "domain": DOMAIN,
+            "type": "is_off",
+            "device_id": device_entry.id,
+            "entity_id": f"{DOMAIN}.test_5678",
+        },
+        {
+            "condition": "device",
+            "domain": DOMAIN,
+            "type": "is_on",
+            "device_id": device_entry.id,
+            "entity_id": f"{DOMAIN}.test_5678",
+        },
+    ]
+    conditions = await async_get_device_automations(hass, "condition", device_entry.id)
+    assert_lists_same(conditions, expected_conditions)
+
+
+async def test_if_state(hass, calls):
+    """Test for turn_on and turn_off conditions."""
+    hass.states.async_set(
+        "humidifier.entity", STATE_ON, {const.ATTR_MODE: const.MODE_AWAY}
+    )
+
+    assert await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)